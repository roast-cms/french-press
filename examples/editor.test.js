import "jest-styled-components"

import React from "react"

import {shallow} from "enzyme"

import {Editor} from "./Editor"

test("Snapshot", () => {
  expect(shallow(<Editor />)).toMatchSnapshot()
})
