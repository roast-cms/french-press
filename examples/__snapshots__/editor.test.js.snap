// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Editor />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "background": "#eee",
              "color": "#999",
              "padding": ".5em",
            }
          }
        >
          Draft Saved.
        </div>,
        <FrenchPress
          callbackError={[Function]}
          callbackStatus={[Function]}
          components={
            Object {
              "Link": null,
              "Picture": [Function],
            }
          }
          controls={
            Object {
              "CancelHeader": [Function],
              "MakeBold": [Function],
              "MakeHeader": [Function],
              "MakeItalic": [Function],
              "MakeLink": [Function],
              "MakeQuote": [Function],
              "UploadImage": [Function],
            }
          }
          editorRef={[Function]}
          options={
            Object {
              "domain": "localhost:3002",
              "imagePlaceholder": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
            }
          }
          placeholder="Write here..."
          slatePlugins={
            Array [
              Object {
                "onKeyDown": [Function],
              },
            ]
          }
          value={
            Object {
              "document": Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Paragraph test.",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "italic",
                              },
                            ],
                            "object": "leaf",
                            "text": "Inline",
                          },
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": " ",
                          },
                          Object {
                            "marks": Array [
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "bold",
                              },
                            ],
                            "object": "leaf",
                            "text": "marks",
                          },
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": " ",
                          },
                          Object {
                            "marks": Array [
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "bold",
                              },
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "italic",
                              },
                            ],
                            "object": "leaf",
                            "text": "test",
                          },
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": ".",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Links ",
                          },
                        ],
                        "object": "text",
                      },
                      Object {
                        "data": Object {
                          "href": "https://github.com/roast-cms/french-press-editor",
                        },
                        "isVoid": false,
                        "nodes": Array [
                          Object {
                            "leaves": Array [
                              Object {
                                "marks": Array [],
                                "object": "leaf",
                                "text": "test",
                              },
                            ],
                            "object": "text",
                          },
                        ],
                        "object": "inline",
                        "type": "link",
                      },
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": ".",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Header test.",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "heading",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Quote test.",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "quote",
                  },
                  Object {
                    "data": Object {
                      "src": "https://raw.githubusercontent.com/roast-cms/french-press-editor/develop/graphics/logo.gif",
                    },
                    "isVoid": true,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "image",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                ],
                "object": "document",
              },
              "object": "value",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Draft Saved.",
          "style": Object {
            "background": "#eee",
            "color": "#999",
            "padding": ".5em",
          },
        },
        "ref": null,
        "rendered": "Draft Saved.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "callbackError": [Function],
          "callbackStatus": [Function],
          "components": Object {
            "Link": null,
            "Picture": [Function],
          },
          "controls": Object {
            "CancelHeader": [Function],
            "MakeBold": [Function],
            "MakeHeader": [Function],
            "MakeItalic": [Function],
            "MakeLink": [Function],
            "MakeQuote": [Function],
            "UploadImage": [Function],
          },
          "editorRef": [Function],
          "options": Object {
            "domain": "localhost:3002",
            "imagePlaceholder": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
          },
          "placeholder": "Write here...",
          "slatePlugins": Array [
            Object {
              "onKeyDown": [Function],
            },
          ],
          "value": Object {
            "document": Object {
              "data": Object {},
              "nodes": Array [
                Object {
                  "data": Object {},
                  "isVoid": false,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": "Paragraph test.",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "paragraph",
                },
                Object {
                  "data": Object {},
                  "isVoid": false,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [
                            Object {
                              "data": Object {},
                              "object": "mark",
                              "type": "italic",
                            },
                          ],
                          "object": "leaf",
                          "text": "Inline",
                        },
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": " ",
                        },
                        Object {
                          "marks": Array [
                            Object {
                              "data": Object {},
                              "object": "mark",
                              "type": "bold",
                            },
                          ],
                          "object": "leaf",
                          "text": "marks",
                        },
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": " ",
                        },
                        Object {
                          "marks": Array [
                            Object {
                              "data": Object {},
                              "object": "mark",
                              "type": "bold",
                            },
                            Object {
                              "data": Object {},
                              "object": "mark",
                              "type": "italic",
                            },
                          ],
                          "object": "leaf",
                          "text": "test",
                        },
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": ".",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "paragraph",
                },
                Object {
                  "data": Object {},
                  "isVoid": false,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": "Links ",
                        },
                      ],
                      "object": "text",
                    },
                    Object {
                      "data": Object {
                        "href": "https://github.com/roast-cms/french-press-editor",
                      },
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "test",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "inline",
                      "type": "link",
                    },
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": ".",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "paragraph",
                },
                Object {
                  "data": Object {},
                  "isVoid": false,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": "Header test.",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "heading",
                },
                Object {
                  "data": Object {},
                  "isVoid": false,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": "Quote test.",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "quote",
                },
                Object {
                  "data": Object {
                    "src": "https://raw.githubusercontent.com/roast-cms/french-press-editor/develop/graphics/logo.gif",
                  },
                  "isVoid": true,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": "",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "image",
                },
                Object {
                  "data": Object {},
                  "isVoid": false,
                  "nodes": Array [
                    Object {
                      "leaves": Array [
                        Object {
                          "marks": Array [],
                          "object": "leaf",
                          "text": "",
                        },
                      ],
                      "object": "text",
                    },
                  ],
                  "object": "block",
                  "type": "paragraph",
                },
              ],
              "object": "document",
            },
            "object": "value",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "background": "#eee",
                "color": "#999",
                "padding": ".5em",
              }
            }
          >
            Draft Saved.
          </div>,
          <FrenchPress
            callbackError={[Function]}
            callbackStatus={[Function]}
            components={
              Object {
                "Link": null,
                "Picture": [Function],
              }
            }
            controls={
              Object {
                "CancelHeader": [Function],
                "MakeBold": [Function],
                "MakeHeader": [Function],
                "MakeItalic": [Function],
                "MakeLink": [Function],
                "MakeQuote": [Function],
                "UploadImage": [Function],
              }
            }
            editorRef={[Function]}
            options={
              Object {
                "domain": "localhost:3002",
                "imagePlaceholder": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
              }
            }
            placeholder="Write here..."
            slatePlugins={
              Array [
                Object {
                  "onKeyDown": [Function],
                },
              ]
            }
            value={
              Object {
                "document": Object {
                  "data": Object {},
                  "nodes": Array [
                    Object {
                      "data": Object {},
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "Paragraph test.",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "paragraph",
                    },
                    Object {
                      "data": Object {},
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [
                                Object {
                                  "data": Object {},
                                  "object": "mark",
                                  "type": "italic",
                                },
                              ],
                              "object": "leaf",
                              "text": "Inline",
                            },
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": " ",
                            },
                            Object {
                              "marks": Array [
                                Object {
                                  "data": Object {},
                                  "object": "mark",
                                  "type": "bold",
                                },
                              ],
                              "object": "leaf",
                              "text": "marks",
                            },
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": " ",
                            },
                            Object {
                              "marks": Array [
                                Object {
                                  "data": Object {},
                                  "object": "mark",
                                  "type": "bold",
                                },
                                Object {
                                  "data": Object {},
                                  "object": "mark",
                                  "type": "italic",
                                },
                              ],
                              "object": "leaf",
                              "text": "test",
                            },
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": ".",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "paragraph",
                    },
                    Object {
                      "data": Object {},
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "Links ",
                            },
                          ],
                          "object": "text",
                        },
                        Object {
                          "data": Object {
                            "href": "https://github.com/roast-cms/french-press-editor",
                          },
                          "isVoid": false,
                          "nodes": Array [
                            Object {
                              "leaves": Array [
                                Object {
                                  "marks": Array [],
                                  "object": "leaf",
                                  "text": "test",
                                },
                              ],
                              "object": "text",
                            },
                          ],
                          "object": "inline",
                          "type": "link",
                        },
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": ".",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "paragraph",
                    },
                    Object {
                      "data": Object {},
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "Header test.",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "heading",
                    },
                    Object {
                      "data": Object {},
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "Quote test.",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "quote",
                    },
                    Object {
                      "data": Object {
                        "src": "https://raw.githubusercontent.com/roast-cms/french-press-editor/develop/graphics/logo.gif",
                      },
                      "isVoid": true,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "image",
                    },
                    Object {
                      "data": Object {},
                      "isVoid": false,
                      "nodes": Array [
                        Object {
                          "leaves": Array [
                            Object {
                              "marks": Array [],
                              "object": "leaf",
                              "text": "",
                            },
                          ],
                          "object": "text",
                        },
                      ],
                      "object": "block",
                      "type": "paragraph",
                    },
                  ],
                  "object": "document",
                },
                "object": "value",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Draft Saved.",
            "style": Object {
              "background": "#eee",
              "color": "#999",
              "padding": ".5em",
            },
          },
          "ref": null,
          "rendered": "Draft Saved.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "callbackError": [Function],
            "callbackStatus": [Function],
            "components": Object {
              "Link": null,
              "Picture": [Function],
            },
            "controls": Object {
              "CancelHeader": [Function],
              "MakeBold": [Function],
              "MakeHeader": [Function],
              "MakeItalic": [Function],
              "MakeLink": [Function],
              "MakeQuote": [Function],
              "UploadImage": [Function],
            },
            "editorRef": [Function],
            "options": Object {
              "domain": "localhost:3002",
              "imagePlaceholder": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
            },
            "placeholder": "Write here...",
            "slatePlugins": Array [
              Object {
                "onKeyDown": [Function],
              },
            ],
            "value": Object {
              "document": Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Paragraph test.",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "italic",
                              },
                            ],
                            "object": "leaf",
                            "text": "Inline",
                          },
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": " ",
                          },
                          Object {
                            "marks": Array [
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "bold",
                              },
                            ],
                            "object": "leaf",
                            "text": "marks",
                          },
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": " ",
                          },
                          Object {
                            "marks": Array [
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "bold",
                              },
                              Object {
                                "data": Object {},
                                "object": "mark",
                                "type": "italic",
                              },
                            ],
                            "object": "leaf",
                            "text": "test",
                          },
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": ".",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Links ",
                          },
                        ],
                        "object": "text",
                      },
                      Object {
                        "data": Object {
                          "href": "https://github.com/roast-cms/french-press-editor",
                        },
                        "isVoid": false,
                        "nodes": Array [
                          Object {
                            "leaves": Array [
                              Object {
                                "marks": Array [],
                                "object": "leaf",
                                "text": "test",
                              },
                            ],
                            "object": "text",
                          },
                        ],
                        "object": "inline",
                        "type": "link",
                      },
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": ".",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Header test.",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "heading",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "Quote test.",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "quote",
                  },
                  Object {
                    "data": Object {
                      "src": "https://raw.githubusercontent.com/roast-cms/french-press-editor/develop/graphics/logo.gif",
                    },
                    "isVoid": true,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "image",
                  },
                  Object {
                    "data": Object {},
                    "isVoid": false,
                    "nodes": Array [
                      Object {
                        "leaves": Array [
                          Object {
                            "marks": Array [],
                            "object": "leaf",
                            "text": "",
                          },
                        ],
                        "object": "text",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                ],
                "object": "document",
              },
              "object": "value",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
